src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/SourceRepositoryExtensions.cs

```diff
@@ -227,6 +229,21 @@ public static async Task<IPackageSearchMetadata> GetPackageMetadataForIdentityAs
             }
         }
 
+        public static async Task<GetVulnerabilityInfoResult> GetVulnerabilityInfoAsync(this SourceRepository sourceRepository, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            var vulnerabilityResource = await sourceRepository.GetResourceAsync<IVulnerabilityInfoResource>(cancellationToken);
+            if (vulnerabilityResource is null)
+            {
+                return null;
+            }
+
+            using (var sourceCacheContext = new SourceCacheContext())
```

Comment:
Share the SourceCacheContext among source repositories.

Reply:
The static extension methods operate on a single source repository. In order to share SourceCacheContext among various source repositories, we would need to supply the SourceCacheContext along with each of the method calls in the SourceRepositoryExtensions. The current precedent set by this static class and its methods is to create a new SourceCacheContext per each call, which is consistent with what we're doing here.

An alternative design would be to store a static reference to the SourceCacheContext but then we'd need methods on this static class that manages the lifecycle of the SourceCacheContext. Ideally, we would have a better way to manage a collection of source repositories and their operations and share the SourceCacheContext across multiple repos.

I'm making note of your suggestion as an improvement that should be made, but it's out of scope of this feature as this simply follows an existing pattern using an existing mechanism which can be improved upon, but was not introduced in this change.

Reply:
> The current precedent set by this static class and its methods is to create a new SourceCacheContext per each call, which is consistent with what we're doing here.

I see. That's a poor pattern. See https://github.com/NuGet/NuGet.Client/commit/9ea2defeabc6ab8687f11298e314c197e95a6b8f. 
The caching context should be shared so that any caching decisions don't need to be changed in 10 different places. 
I think we should change that, I've created an issue for that https://github.com/NuGet/Home/issues/12859. 


> simply follows an existing pattern using an existing mechanism which can be improved upon, but was not introduced in this change.

I don't think we should be propagating bad patterns, even if it's a preexisting pattern. However, I think it's reasonable to say that all of those discussions should be handled on a case by case basis, as the cost/impact is different for each one individually.

My suggestion is to just pass down the source cache context, but I'm not too concerned about this single change, so feel free to make a call.

Reply:
@nkolev92 thanks for creating the follow-up issue!